source/event_stream.c

```diff
@@ -0,0 +1,1118 @@
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int
+aws_event_stream_add_int16_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int16_t value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = sizeof(value),
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_INT16,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    aws_write_u16(header.header_value.static_val, (uint16_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int
+aws_event_stream_add_int32_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int32_t value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = sizeof(value),
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_INT32,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    aws_write_u32(header.header_value.static_val, (uint32_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int
+aws_event_stream_add_int64_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int64_t value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = sizeof(value),
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_INT64,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    aws_write_u64(header.header_value.static_val, (uint64_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int aws_event_stream_add_bytebuf_header(struct aws_array_list *headers, const char *name, uint8_t name_len,
+                                        uint8_t *value, uint16_t value_len, int8_t copy) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = value_len,
+            .value_owned = copy,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_BYTE_BUF
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+
+    if (copy) {
+        header.header_value.variable_len_val = aws_mem_acquire(headers->alloc, value_len);
+        if (!header.header_value.variable_len_val) {
+            return aws_raise_error(AWS_ERROR_OOM);
+        }
+
+        header.value_owned = 1;
+        memcpy((void *) header.header_value.variable_len_val, (void *) value, value_len);
+    } else {
+        header.value_owned = 0;
+        header.header_value.variable_len_val = (const uint8_t *) value;
+    }
+
+    if (aws_array_list_push_back(headers, (void *) &header)) {
```

Comment:
missing `&& copy`

Reply:
? push back doesn't take a double pointer.

Reply:
no && as in logical and. 
if(copy) is false then no allocation